src/runtime/pymodule.cs

```diff
@@ -37,5 +40,40 @@ public static PyModule FromString(string name, string code)
             PythonException.ThrowIfIsNull(m);
             return new PyModule(ref m);
         }
+
+        public static PyModule Create(string name, string filename = "none")
```

Comment:
This method violates single responsibility principle.

First, `Create` should never return an existing module. Instead, the caller should be able to check if module already exists by name by calling a separate function.

Second, I an uncertain about setting up `__builtins__` by default. It would be great to know why Python does not do it before discarding that potentially good rationale. Users can always manually do `newModule.Dict.Set("__builtins__", GetBuiltIns())`. If this is a common scenario in your product, just make a factory method combining the two.

Third, I don't think adding new module to `sys.modules` right away is a good idea. Especially, overwriting the existing value. This might cause hard to debug surprise issues.

Reply:
> Second, I am uncertain about setting up `__builtins__` by default. It would be great to know why Python does not do it before discarding that potentially good rationale. Users can always manually do `newModule.Dict.Set("__builtins__", GetBuiltIns())`. If this is a common scenario in your product, just make a factory method combining the two.
> 

I didn't see a public method that would allow access to the __builtins__. There is an internal method, on `Runtime`, but not a public method that could be used outside.